portfolio/src/main/java/com/google/sps/servlets/UserStatusServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/home")
+public class UserStatusServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+    PrintWriter out = response.getWriter();
+    
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      String loginUrl = userService.createLoginURL("/home");
+      out.println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
+      return;
+    }
+    else{
+        String userEmail = userService.getCurrentUser().getEmail();
+        String logoutUrl = userService.createLogoutURL("/index.html");
+        response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+        response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    }
+  }
+
```

Comment:
If you want to improve this in the future, instead of returning HTML you can just return JSON and use javascript with an asynchronous call to fetch the loginURL data, and in the case the user is logged in, userEmail and logoutURL.